Examples/BasketLosses/BasketLosses.cpp

```diff
@@ -70,7 +70,7 @@ int main(int, char* []) {
         for(Size i=0; i<hazardRates.size(); i++)
             names.push_back(std::string("Acme") + std::to_string(i));
         std::vector<Handle<DefaultProbabilityTermStructure> > defTS;
-        for (double& hazardRate : hazardRates) {
+        for (Real& hazardRate : hazardRates) {
```

Comment:
Why not use `auto` here instead of `Real`? Generally it's better to use `auto` wherever possible to immunize the code against unintentional datatype conversions after refactoring. Even better would be `const auto` since the value is not modified.

Also in this case it's better to declare the loop variable as a value instead of a reference because it's cheap to copy. (Even if the compiler optimizes out the deference it's still clearer to declare as a value).

Reply:
I generally agree with that, though this would apply to many places throughout the codebase. In this PR, it's minimum changes to replace `double` with `Real`.

Perhaps we should tackle that as a separate set of changes / PR - review and use `auto` wherever it makes sense. I wouldn't over-use it, as sometimes it impacts readability when the type isn't clear from the context. @lballabio - what's your take on that?

Reply:
I don't have a strong opinion about `auto` if the type is simple.  There are a dozen places in the PR when `Real` could be `Time` or `Rate`, too, but the current changes an improvement over `double` anyway.  The rest can come later (or not at all).

Reply:
Ok - we take it that there is no further action needed on this point.